### ISSUE TITLE
regression with `__new__` and/or `Self`, v 1.1.295

### ISSUE THREAD

**Issue by arjenzorgdoc** [microsoft/pyright#4705]
test.py:
```python
class X(str):
    pass

x: X = str.__new__(X, "y")
print(x, type(x))
```

```
$ npx pyright@1.1.294 test.py 
No configuration file found.
No pyproject.toml file found.
stubPath /home/arjen/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.294
0 errors, 0 warnings, 0 informations 
Completed in 0.58sec
```

```
$ npx pyright@1.1.295 test.py 
No configuration file found.
No pyproject.toml file found.
stubPath /home/arjen/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.295
/home/arjen/test.py
  /home/arjen/test.py:4:8 - error: Expression of type "Self@str" cannot be assigned to declared type "X"
    "str" is incompatible with "X" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.61sec
```

**Comment by arjenzorgdoc** [comment]
```
$ npx pyright@1.1.296 test.py 
Need to install the following packages:
  pyright@1.1.296
Ok to proceed? (y) y
No configuration file found.
No pyproject.toml file found.
stubPath /home/arjen/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.296
/home/arjen/test.py
  /home/arjen/test.py:4:8 - error: Expression of type "Self@str" cannot be assigned to declared type "X"
    "str" is incompatible with "X" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.601sec
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.297, which I just published. It will also be included in a future release of pylance.
